cpp/dolfinx/refinement/refine.h

```diff
@@ -19,44 +20,28 @@
 
 namespace dolfinx::refinement
 {
-namespace impl
-{
-/// @brief create the refined mesh by optionally redistributing it
-template <std::floating_point T>
-mesh::Mesh<T>
-create_refined_mesh(const mesh::Mesh<T>& mesh,
-                    const graph::AdjacencyList<std::int64_t>& cell_adj,
-                    std::span<const T> new_vertex_coords,
-                    std::array<std::size_t, 2> xshape, bool redistribute,
-                    mesh::GhostMode ghost_mode)
-{
-  if (dolfinx::MPI::size(mesh.comm()) == 1)
-  {
-    // No parallel construction necessary, i.e. redistribute also has no
-    // effect
-    return mesh::create_mesh(mesh.comm(), cell_adj.array(),
-                             mesh.geometry().cmap(), new_vertex_coords, xshape,
-                             ghost_mode);
-  }
-  else
-  {
-    // Let partition handle the parallel construction of the mesh
-    return partition<T>(mesh, cell_adj, new_vertex_coords, xshape, redistribute,
-                        ghost_mode);
-  }
-}
-} // namespace impl
+/// @brief Partitioner that maintains the regional distribution for a
+/// refined mesh. Meaning, process local data is just maintained and no
+/// redistribution happens.
+///
+/// @param[in] comm MPI Communicator
+/// @param[in] nparts Number of partitions (input has no effect)
+/// @param[in] cell_types Cell types in the mesh
+/// @param[in] cells Lists of cells of each cell type.
+/// @return Destination ranks for each cell on this process
+graph::AdjacencyList<std::int32_t>
+maintain_coarse_partitioner(MPI_Comm comm, int nparts,
```

Comment:
I think this name could be improved - maybe 'identity_partitioner'?

Reply:
Open for new names, I agree it is not nicely named. 

I wasn't entirely sure if this partitioner routine makes any sense outside of the refinement context (which the name kind of hints). https://github.com/FEniCS/dolfinx/issues/3358 might be a related to this.

Reply:
I don't think we need it - we can pass `nullptr` as the partitioning function, in which case `mesh::create_mesh` doesn't do any redistribution. Or am I missing something?